libs/designsystem/src/lib/components/icon/icon.component.ts

```diff
@@ -54,13 +57,33 @@ export class IconComponent implements OnChanges {
     }
   }
 
-  constructor(@Optional() @Inject(ICON_SETTINGS) private iconSettings?: IconSettings) {}
+  /**
+   * iconRegistryService: Registry for custom icons.
+   * iconSettings: @deprecated Use KirbyIconRegistryService for adding custom icons.
+   */
+  constructor(
+    private iconRegistryService: KirbyIconRegistryService,
+    @Optional() @Inject(ICON_SETTINGS) private iconSettings?: IconSettings
+  ) {
+    this.combineIconSettings();
+  }
+
+  private combineIconSettings() {
+    const iconSettings: Icon[] =
+      this.iconSettings && this.iconSettings.icons
```

Comment:
You can just:

`
let iconSettingsIcons = this.iconSettings?.icons;
if(!icons) {
iconSettingsIcons = [this.defaultIcon]
}
`

Reply:
@lydemann Nope, we are on TypeScript 3.6.5 for compability reasons